components/brave_wallet/browser/brave_wallet_provider.cc

```diff
@@ -0,0 +1,42 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/brave_wallet_provider.h"
+
+#include <utility>
+
+#include "brave/components/brave_wallet/brave_wallet_service.h"
+
+namespace brave_wallet {
+
+BraveWalletProvider::BraveWalletProvider(BraveWalletService* wallet_service)
```

Comment:
This class seems like an unnecessary extra layer. I think it would be simpler for BraveWalletService to just implement `brave_wallet::mojom::BraveWalletProvider` directly